js/components/sidebar.vue

```diff
@@ -83,6 +83,15 @@ export default {
                 };
             });
         }
+    },
+    methods: {
+        onSearch() {
+            const terms = this.$els.terms.value;
+            if (terms && terms.length > 2) {
+                this.$go({name: 'search', query: { terms: terms }});
+                this.$root.me.$emit('updated');
```

Comment:
OK, so I think you can remote this emit and just update on route change in search.vue.
I will comment in search.vue